moto/awslambda/responses.py

```diff
@@ -397,7 +397,15 @@ def _delete_function(self) -> TYPE_RESPONSE:
         return 204, {}, ""
 
     @staticmethod
-    def _set_configuration_qualifier(configuration: Dict[str, Any], qualifier: str) -> Dict[str, Any]:  # type: ignore[misc]
+    def _set_configuration_qualifier(configuration: Dict[str, Any], function_name: str, qualifier: str) -> Dict[str, Any]:  # type: ignore[misc]
```

Comment:
I'm not sure I understand the original intent of injecting the qualifier into the configuration here, but I simply extended it to fetch the qualifier from the name itself if present.